### ISSUE TITLE
Subpixel anti-aliasing failing on tiles with very specific last lines

### ISSUE THREAD

**Issue by burodepeper** [atom/atom#8620]
### Steps to reproduce
1. Create a new Markdown document with the content below
2. Use mentioned syntax theme and typeface
3. Notice the color difference between quote 10 and 11
### Cause

Normally, with my settings, all lines are 21px high, and normal text is rendered with a height of 16px. In the example however, the #-sign, scoped as `markup heading marker gfm`, whose `font-weight: bold` inherited from being a heading, is reset to `font-weight: normal`. This causes the #-sign to be rendered as **17px** high, making the line itself **22px** instead of 21px, and pushing a single pixel outside the tile, causing the entire tile to lose subpixel anti-aliasing. Furthermore, as far as I have been able to deduce, this only happens when the `font-weight` is reset from `bold` to `normal`, although this could be a coincidence of course.

Heads up to @as-cii who I've already discussed this with on Slack. And can someone give me a detective's badge, cause I certainly feel like one! : )
### Screenshot

![schermafbeelding 2015-09-01 om 15 48 14](https://cloud.githubusercontent.com/assets/3703530/9605943/e8ab3138-50c0-11e5-9bb2-30ecbdb65a5a.png)

Notice how the tile is 1px higher than the gutter-tile next to it.
### Sample code

``` markdown
> Awesome quote 1
> Awesome quote 2
> Awesome quote 3
> Awesome quote 4
> Awesome quote 5
> Awesome quote 6
> Awesome quote 7
> Awesome quote 8
> Awesome quote 9
> Awesome quote 10

# I am a header that's screwing things up

> Awesome quote 11
```
### Environment

**Atom**: 1.0.9
**Theme**: https://atom.io/packages/minimal-syntax
**Typeface**: https://github.com/chrissimpkins/Hack
**font-size**: 14px
**line-height**: 1.5

**Comment by burodepeper** [comment]
![schermafbeelding 2015-09-05 om 09 45 44](https://cloud.githubusercontent.com/assets/3703530/9698368/ee62c5f6-53b2-11e5-8cd5-54dbf2104296.png)

The typeface in this example is set to `Menlo` to take `Hack` out of the equation. The culprit here is the 2px red outline I've given the `FIXME`, which does _not_ render any pixels outside of the tile.

**Comment by burodepeper** [comment]
Hey @as-cii, I've also contacted the author of the typeface and submitted an issue at his repo, as you can see above.

I've found that if I add
```less
atom-text-editor::shadow {
  .line {
    height: 21px;
    overflow: hidden;
  }
}
```
to my `styles.less`, that this patches the issue. Of course, the `21px` is a variable that depends on the font-size and line-height in my settings, but is there a particular reason that `.line` doesn't have a `height` and `overflow:hidden` set in stone?

**Comment by burodepeper** [comment]
A new version of the Hack typeface has been released, and it has fixed my initial issue.
However, I've been digging a little further, and I now do understand the why of the issue. Is there any hope of forcing Chrome to use subpixel-aa? I've been experimenting with adding a 2px outline to search results, which of course triggers this bug, especially since tiles now consist of 6 lines.

**Comment by as-cii** [comment]
I think the only way to solve this once for all is to switch back to a single rendering layer, so that subpixel anti-aliasing is not a problem anymore: tiling is just awesome in terms of painting times, but we have underestimated its negative impact on text rendering.

We have already made a bunch of experiments and I think it will land on Atom at some point in the future, so stay tuned! And thanks for your super helpful feedback!
